src/ServiceControl.Transports.PostgreSql/PostgreSqlTransportCustomization.cs

```diff
@@ -0,0 +1,86 @@
+﻿namespace ServiceControl.Transports.PostgreSql;
+
+using System.Linq;
+using System.Runtime.CompilerServices;
+using BrokerThroughput;
+using Microsoft.Extensions.DependencyInjection;
+using NServiceBus;
+using NServiceBus.Logging;
+using NServiceBus.Transport.PostgreSql;
+
+public class PostgreSqlTransportCustomization : TransportCustomization<PostgreSqlTransport>
+{
+    protected override void CustomizeTransportForPrimaryEndpoint(EndpointConfiguration endpointConfiguration, PostgreSqlTransport transportDefinition, TransportSettings transportSettings) =>
+        transportDefinition.TransportTransactionMode = TransportTransactionMode.SendsAtomicWithReceive;
+
+    protected override void CustomizeTransportForAuditEndpoint(EndpointConfiguration endpointConfiguration, PostgreSqlTransport transportDefinition, TransportSettings transportSettings) =>
+        transportDefinition.TransportTransactionMode = TransportTransactionMode.ReceiveOnly;
+
+    protected override void CustomizeTransportForMonitoringEndpoint(EndpointConfiguration endpointConfiguration, PostgreSqlTransport transportDefinition, TransportSettings transportSettings) =>
+        transportDefinition.TransportTransactionMode = TransportTransactionMode.ReceiveOnly;
+
+    protected override void AddTransportForPrimaryCore(IServiceCollection services,
+        TransportSettings transportSettings) =>
+        services.AddSingleton<IBrokerThroughputQuery, PostgreSqlQuery>();
+
+    protected override void AddTransportForMonitoringCore(IServiceCollection services, TransportSettings transportSettings)
+    {
+        services.AddSingleton<IProvideQueueLength, QueueLengthProvider>();
+        services.AddHostedService(provider => provider.GetRequiredService<IProvideQueueLength>());
+    }
+
+    protected override PostgreSqlTransport CreateTransport(TransportSettings transportSettings, TransportTransactionMode preferredTransactionMode = TransportTransactionMode.ReceiveOnly)
+    {
+        var connectionString = transportSettings.ConnectionString.RemoveCustomConnectionStringParts(out var customSchema, out var subscriptionsTableSetting);
+
+        var transport = new PostgreSqlTransport(connectionString);
+
+        var subscriptions = transport.Subscriptions;
+
+        if (customSchema != null)
+        {
+            transport.DefaultSchema = customSchema;
+            subscriptions.SubscriptionTableName = new SubscriptionTableName(DefaultSubscriptionTableName, customSchema);
+        }
+
+        if (subscriptionsTableSetting != null)
+        {
+            var subscriptionsAddress = QueueAddress.Parse(subscriptionsTableSetting);
+
+            subscriptions.SubscriptionTableName =
+                new SubscriptionTableName(subscriptionsAddress.Table,
+                    subscriptionsAddress.Schema ?? customSchema);
+        }
+
+        if (transportSettings.GetOrDefault<bool>("TransportSettings.EnableDtc"))
```

Comment:
not originating from this PR I know, but why is this not a string constant somewhere?

Reply:
Well it's only used once so probably didn't see a point to it?